core/src/main/java/cocktailofferer/service/image/ImageMetadata.java

```diff
@@ -0,0 +1,28 @@
+package cocktailofferer.service.image;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.*;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Getter
+@EqualsAndHashCode(of = "id")
+public class ImageMetadata {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @NotNull
+    @Size(min = ImageMetadataConstraints.Key.MIN_SIZE, max = ImageMetadataConstraints.Key.MAX_SIZE)
+    @Column(unique = true)
+    private String imageKey;
+
+    @NotNull
+    @Size(min = ImageMetadataConstraints.Name.MIN_SIZE, max = ImageMetadataConstraints.Name.MAX_SIZE)
+    private String name;
```

Comment:
Do we really need a name? Is it going to be helpful?

Reply:
for downloading with original name

Reply:
Is there such a requirement? What is the use case?

Since names of files are given by users they may be confusing, strange, meaningless, may contain abusive words, etc. So, it's not always good idea to store original filenames.

Reply:
```suggestion
    private String originalFilename;
```